tensorbay/client/requests.py

```diff
@@ -157,6 +157,9 @@ def request(  # type: ignore[override]  # pylint: disable=signature-differs
                 logger.error(
                     "Unexpected status code(%d)!%s", response.status_code, ResponseLogging(response)
                 )
+                error_code = response.json()["code"]
```

Comment:
The `response.json()` needs to be called to get the `error_code`, this is a redundant calculation due to it will be called again outside this function in most cases.

Reply:
The statement is only called when an exception occurs.